### PR TITLE
Added caching section

### LINKED ISSUE #12521: [ML.NET] Include Cache in HOW TOs at docs.microsoft.com for ML.NET
It’d be good to have a short HOW TO page explaining when to use cache, when not to use it, pros/cons, and the simple line of code on how to use it in the pipeline.

This is almost the only place in the reference talking about it, but it is not very clear:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.learningpipelineextensions.appendcachecheckpoint?view=ml-dotnet#Microsoft_ML_LearningPipelineExtensions_AppendCacheCheckpoint__1_Microsoft_ML_IEstimator___0__Microsoft_ML_Runtime_IHostEnvironment_

A HOW TO page should say that for small/medium datasets and algorithms that do multiple passes it improves performance, but for large datasets it can impact negatively since you might not have enough memory for loading the whole dataset and when the dataset is large it is better to read from the file streaming chunks of data as needed when training (automatically done by the reader).

### PR DESCRIPTION
**Pull request by luisquintanilla**
## Summary

Add information on `Cache` and `AppendCacheCheckpoint`.

Preview Link: https://review.docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/train-machine-learning-model-ml-net?branch=pr-en-us-17228#caching-data

Fixes #12521

[AB#1681079](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1681079)

### CODE REVIEW

File: docs/machine-learning/how-to-guides/train-machine-learning-model-ml-net.md [hunk 1]
```diff
@@ -119,17 +119,49 @@ IDataView transformedTrainingData = dataPrepTransformer.Transform(trainData);
...
+By using the [`Concatenate`](xref:Microsoft.ML.TransformExtensionsCatalog.Concatenate%2A) method during pre-processing to create a new column called `Features`, there is no need to specify the feature column name in the parameters of the algorithm since it already exists in the pre-processed `IDataView`. The label column is `CurrentPrice`, but since the [`ColumnName`](xref:Microsoft.ML.Data.ColumnNameAttribute) attribute is used in the data model, ML.NET renames the `CurrentPrice` column to `Label` which removes the need to provide the `labelColumnName` parameter to the machine learning algorithm estimator.
 
 If you don't want to use the default column names, pass in the names of the feature and label columns as parameters when defining the machine learning algorithm estimator as demonstrated by the subsequent snippet:
 
 ```csharp
 var UserDefinedColumnSdcaEstimator = mlContext.Regression.Trainers.Sdca(labelColumnName: "MyLabelColumnName", featureColumnName: "MyFeatureColumnName");
 ```
 
+## Caching data
```
Comment (natke) [inline docs/machine-learning/how-to-guides/train-machine-learning-model-ml-net.md]: Looks great! Could we add a line or two that would help users understand when they would want to cache outside of the estimator chain and when they would want to cache the IDataView directly?

Comment (luisquintanilla) [reply docs/machine-learning/how-to-guides/train-machine-learning-model-ml-net.md]: resolved

**Comment by BillWagner** [comment]
Hey @luisquintanilla @natke I'm going through old PRs. Is this ready? It looks like all the comments have been addressed.

**Comment by luisquintanilla** [comment]
@BillWagner working on it at the moment. Should be merging it by EOD. Thanks.

**Review by natke — APPROVED** [review]
LGTM :shipit: !

**Pull request merged**